### PR TITLE
EIP-1559 - Provide edit button for transaction screens to update gas

### PR DESCRIPTION
**Pull request by darkwing**
This provides an Edit button for transaction screens so that the user may update gas.  It doesn't handle the submission or change of that gas, much of which is done in the "Speed Up" PR:  https://github.com/MetaMask/metamask-extension/pull/11510

### CODE REVIEW

File: ui/pages/confirm-transaction-base/confirm-transaction-base.component.js [hunk 1]
```diff
@@ -346,10 +346,19 @@ export default class ConfirmTransactionBase extends Component {
       </div>
     ) : null;
 
+    const showInlineControls = process.env.SHOW_EIP_1559_UI
```
Comment (brad-decker) [inline ui/pages/confirm-transaction-base/confirm-transaction-base.component.js:349]: I'm losing my mind, I thought I did this already in #11481 -- but apparently this change got lost in the rebase? 

anyways this all looks fine but we should probably remove the work I did there in transaction-detail-item if its not needed now

**Review by brad-decker — CHANGES_REQUESTED** [review]

File: ui/pages/confirm-transaction-base/confirm-transaction-base.component.js (hunk 1 above)
Comment (darkwing) [reply ui/pages/confirm-transaction-base/confirm-transaction-base.component.js:349]: Gah, sorry!  I thought so as well but didn't think to look in the individual `item` component -- sorry!

Comment (brad-decker) [reply ui/pages/confirm-transaction-base/confirm-transaction-base.component.js:349]: No worries the rebase issue was most likely my fault :D

**Review by brad-decker — DISMISSED** [review]

**Review by ryanml — DISMISSED** [review]

**Review by ryanml — APPROVED** [review]

File: ui/components/app/transaction-detail/transaction-detail.component.js [hunk 2]
```diff
@@ -1,15 +1,31 @@
...
+export default function TransactionDetail({ rows = [], onEdit }) {
+  const t = useContext(I18nContext);
+
+  return (
+    <div
+      className={classNames('transaction-detail', {
+        'transaction-detail--editable': Boolean(onEdit),
+      })}
+    >
+      {onEdit && (
```
Comment (dan437) [inline ui/components/app/transaction-detail/transaction-detail.component.js:18]: Sweet!

**Review by dan437 — APPROVED** [review]
Looks good to me!

**Pull request merged**